# 05_231211_RePRSTrials

### 主目的

Tonga集団のサンプルを用いて倹約遺伝子仮説を検討する。

1. 過去のPRSを推定する：今回はその前段階として現在ベースでPRSを推定する。
2. ポリジェニックセレクションの検討：これから（まず手法をもう一度チェック！）

### 実行にあたって

Tonga集団のデータを用いてPRSを計算したい。が、いま手元に彼らの身長や体重に関するデータがないことがわかった。

とりあえずその前段階として遺伝統計学夏の学校2023の実習を再現してみる。

### 遺伝統計学夏の学校2023【QC以降の手順確認】

まずはデータファイルをダウンロード・解凍する。

In [2]:
mkdir -p ~/hmiwa1/storage/hmiwa-lab/05_231211_RePRSTrials
cd ~/hmiwa1/storage/hmiwa-lab/05_231211_RePRSTrials
ls

In [3]:
wget http://www.sg.med.osaka-u.ac.jp/files/StatGenSummerSchool2023_GenomeDataAnalysis5.zip
ls

--2023-12-11 17:12:43--  http://www.sg.med.osaka-u.ac.jp/files/StatGenSummerSchool2023_GenomeDataAnalysis5.zip
Resolving www.sg.med.osaka-u.ac.jp (www.sg.med.osaka-u.ac.jp)... 133.1.138.171
Connecting to www.sg.med.osaka-u.ac.jp (www.sg.med.osaka-u.ac.jp)|133.1.138.171|:80... connected.
HTTP request sent, awaiting response... 200 OK
Length: 185582439 (177M) [application/zip]
Saving to: ‘StatGenSummerSchool2023_GenomeDataAnalysis5.zip’

StatGenSummerSchool 100%[===================>] 176.98M  21.6MB/s    in 6.9s    

2023-12-11 17:12:50 (25.8 MB/s) - ‘StatGenSummerSchool2023_GenomeDataAnalysis5.zip’ saved [185582439/185582439]

StatGenSummerSchool2023_GenomeDataAnalysis5.zip


In [6]:
unzip StatGenSummerSchool2023_GenomeDataAnalysis5.zip

Archive:  StatGenSummerSchool2023_GenomeDataAnalysis5.zip
   creating: StatGenSummerSchool2023_GenomeDataAnalysis5/
   creating: StatGenSummerSchool2023_GenomeDataAnalysis5/PRSice/
   creating: StatGenSummerSchool2023_GenomeDataAnalysis5/PRSice/Input/


  inflating: StatGenSummerSchool2023_GenomeDataAnalysis5/PRSice/Input/1KG_EUR_QC.bed  
  inflating: StatGenSummerSchool2023_GenomeDataAnalysis5/PRSice/Input/1KG_EUR_QC.bim  
  inflating: StatGenSummerSchool2023_GenomeDataAnalysis5/PRSice/Input/1KG_EUR_QC.fam  
  inflating: StatGenSummerSchool2023_GenomeDataAnalysis5/PRSice/Input/phenotype_RA.txt  
  inflating: StatGenSummerSchool2023_GenomeDataAnalysis5/PRSice/Input/RA_GWASmeta_European_PRSise.txt.gz  
   creating: StatGenSummerSchool2023_GenomeDataAnalysis5/PRSice/Output/
  inflating: StatGenSummerSchool2023_GenomeDataAnalysis5/PRSice/Output/1KG_EUR-RAGWAS-PRSboxplot.png  
  inflating: StatGenSummerSchool2023_GenomeDataAnalysis5/PRSice/Output/1KG_EUR-RAGWAS.best  
  inflating: StatGenSummerSchool2023_GenomeDataAnalysis5/PRSice/Output/1KG_EUR-RAGWAS.best.txt  
  inflating: StatGenSummerSchool2023_GenomeDataAnalysis5/PRSice/Output/1KG_EUR-RAGWAS.log  
  inflating: StatGenSummerSchool2023_GenomeDataAnalysis5/PRSice/Output/1KG_EUR-RAGWAS.

In [7]:
ls

StatGenSummerSchool2023_GenomeDataAnalysis5
StatGenSummerSchool2023_GenomeDataAnalysis5.zip


In [8]:
ls StatGenSummerSchool2023_GenomeDataAnalysis5

PRSice  StatGenSummerSchool2023_GenomeDataAnalysis5.pdf


In [9]:
ls StatGenSummerSchool2023_GenomeDataAnalysis5/PRSice

Input                      PRsice_Command.txt  PRSice_mac.R
Output                     PRSice_linux        PRSice.R
PRsice_Command_cygwin.txt  PRSice_linux.R      PRSice_win64.exe
PRsice_Command_mac.txt     PRSice_mac


In [10]:
ls StatGenSummerSchool2023_GenomeDataAnalysis5/PRSice/Input
ls StatGenSummerSchool2023_GenomeDataAnalysis5/PRSice/Output

1KG_EUR_QC.bed  1KG_EUR_QC.fam    RA_GWASmeta_European_PRSise.txt.gz
1KG_EUR_QC.bim  phenotype_RA.txt
1KG_EUR-RAGWAS_BARPLOT_2019-08-25.png        1KG_EUR-RAGWAS.log
1KG_EUR-RAGWAS_BARPLOT_2021-07-14.png        1KG_EUR-RAGWAS-PRSboxplot.png
1KG_EUR-RAGWAS.best                          1KG_EUR-RAGWAS.prsice
1KG_EUR-RAGWAS.best.txt                      1KG_EUR-RAGWAS.summary
1KG_EUR-RAGWAS_HIGH-RES_PLOT_2019-08-25.png  1KG_EUR_Sample.txt
1KG_EUR-RAGWAS_HIGH-RES_PLOT_2021-07-14.png  PlotPRSice.R


#### 入力ファイルの確認

In [11]:
ls StatGenSummerSchool2023_GenomeDataAnalysis5/PRSice/Input

1KG_EUR_QC.bed  1KG_EUR_QC.fam    RA_GWASmeta_European_PRSise.txt.gz
1KG_EUR_QC.bim  phenotype_RA.txt


In [15]:
zcat StatGenSummerSchool2023_GenomeDataAnalysis5/PRSice/Input/RA_GWASmeta_European_PRSise.txt.gz | head -n 10

SNP	CHR	BP	A1	A2	P	OR
chr1:751343	1	751343	A	T	0.01	0.85
chr1:751756	1	751756	T	C	0.01	1.17
rs3094315	1	752566	A	G	0.0093	1.14
rs3131972	1	752721	A	G	0.009	0.88
rs3131971	1	752894	T	C	0.01	0.87
chr1:753405	1	753405	A	C	0.01	1.17
chr1:753425	1	753425	T	C	0.0097	0.87
rs2073814	1	753474	G	C	0.009	1.14
rs2073813	1	753541	A	G	0.0095	0.85

gzip: stdout: Broken pipe


In [16]:
cat StatGenSummerSchool2023_GenomeDataAnalysis5/PRSice/Input/phenotype_RA.txt | head -n 10

HG00096	HG00096	2
HG00097	HG00097	1
HG00099	HG00099	1
HG00100	HG00100	1
HG00101	HG00101	2
HG00102	HG00102	2
HG00103	HG00103	1
HG00104	HG00104	1
HG00106	HG00106	1
HG00108	HG00108	1


#### PRS解析の実行

In [4]:
cd ~/hmiwa1/storage/hmiwa-lab/05_231211_RePRSTrials/StatGenSummerSchool2023_GenomeDataAnalysis5/PRSice
mkdir -p Output_2
touch Output_2/.gitkeep
ls

Input     PRsice_Command_cygwin.txt  PRSice_linux.R  PRSice_win64.exe
lib       PRsice_Command_mac.txt     PRSice_mac
Output    PRsice_Command.txt         PRSice_mac.R
Output_2  PRSice_linux               PRSice.R


In [26]:
Rscript PRSice_linux.R \
--prsice PRSice_linux \
--dir . \
--out ./Output_2/1KG_EUR-RAGWAS \
--base ./Input/RA_GWASmeta_European_PRSise.txt.gz \
--target ./Input/1KG_EUR_QC \
--pheno ./Input/phenotype_RA.txt \
--thread 1 \
--stat OR \
--binary-target T

PRSice 2.3.3 (2020-08-05) 
https://github.com/choishingwan/PRSice
(C) 2016-2020 Shing Wan (Sam) Choi and Paul F. O'Reilly
GNU General Public License v3
If you use PRSice in any published work, please cite:
Choi SW, O'Reilly PF.
PRSice-2: Polygenic Risk Score Software for Biobank-Scale Data.
GigaScience 8, no. 7 (July 1, 2019)
2023-12-11 18:02:59
./PRSice_linux \
    --a1 A1 \
    --a2 A2 \
    --bar-levels 0.001,0.05,0.1,0.2,0.3,0.4,0.5,1 \
    --base ./Input/RA_GWASmeta_European_PRSise.txt.gz \
    --binary-target T \
    --bp BP \
    --chr CHR \
    --clump-kb 250kb \
    --clump-p 1.000000 \
    --clump-r2 0.100000 \
    --interval 5e-05 \
    --lower 5e-08 \
    --num-auto 22 \
    --or  \
    --out ./Output_2/1KG_EUR-RAGWAS \
    --pheno ./Input/phenotype_RA.txt \
    --pvalue P \
    --seed 1452368063 \
    --snp SNP \
    --stat OR \
    --target ./Input/1KG_EUR_QC \
    --thread 1 \
    --upper 0.5

Initializing Genotype file: ./Input/1KG_EUR_QC (bed) 

Start processing RA_GWA

#### Output結果の確認

In [27]:
ls Output_2

1KG_EUR-RAGWAS_BARPLOT_2023-12-11.png        1KG_EUR-RAGWAS.log
1KG_EUR-RAGWAS.best                          1KG_EUR-RAGWAS.prsice
1KG_EUR-RAGWAS_HIGH-RES_PLOT_2023-12-11.png  1KG_EUR-RAGWAS.summary


In [28]:
head Output_2/1KG_EUR-RAGWAS.prsice

Pheno	Set	Threshold	R2	P	Coefficient	Standard.Error	Num_SNP
-	Base	5e-08	0.0122466	0.0629842	12.6059	6.77991	257
-	Base	5.005e-05	0.0126982	0.0583659	22.8194	12.055	479
-	Base	0.00010005	0.01382	0.0484141	27.829	14.0998	561
-	Base	0.00015005	0.0139415	0.0474483	30.2877	15.2792	611
-	Base	0.00020005	0.0143518	0.0443525	33.0976	16.4604	658
-	Base	0.00025005	0.0146751	0.0420519	35.1525	17.2909	691
-	Base	0.00030005	0.015445	0.0370909	37.9226	18.1904	729
-	Base	0.00035005	0.0160981	0.0333494	41.2045	19.3644	778
-	Base	0.00040005	0.0162487	0.0325501	43.4408	20.3223	817


In [29]:
head Output_2/1KG_EUR-RAGWAS.summary

Phenotype	Set	Threshold	PRS.R2	Full.R2	Null.R2	Prevalence	Coefficient	Standard.Error	P	Num_SNP
-	Base	0.46	0.0512432	0.0512432	0	-	5081.42	1356.18	0.000179058	101385


In [30]:
#画像表示準備
mkdir -p ~/hmiwa/hmiwa-lab/d_20231023/05_231211_RePRSTrials/pic
cp Output_2/*.png ~/hmiwa/hmiwa-lab/d_20231023/05_231211_RePRSTrials/pic/

【1KG_EUR-RAGWAS_BARPLOT_2023-12-11.png（左）】【1KG_EUR-RAGWAS_BARPLOT_2023-12-11.png（右）】

<img src="pic/1KG_EUR-RAGWAS_BARPLOT_2023-12-11.png" width="49%">
<img src="pic/1KG_EUR-RAGWAS_HIGH-RES_PLOT_2023-12-11.png" width="49%">

#### PRS解析結果の検討

In [31]:
head Output_2/1KG_EUR-RAGWAS.best

FID IID In_Regression PRS
HG00096 HG00096 Yes -0.000627027476
HG00097 HG00097 Yes -0.00058856461
HG00099 HG00099 Yes -0.000713382756
HG00100 HG00100 Yes -0.000571751231
HG00101 HG00101 Yes -0.000530233701
HG00102 HG00102 Yes -0.000583765514
HG00103 HG00103 Yes -0.000573780098
HG00104 HG00104 Yes -0.000506927325
HG00106 HG00106 Yes -0.000537212649


In [55]:
#ファイルの調整
awk '$1!="FID" && $3="Yes" {print $1"\t"$2"\t"$4}' Output_2/1KG_EUR-RAGWAS.best > Output_2/1KG_EUR-RAGWAS.best.txt

In [56]:
cat Output_2/1KG_EUR-RAGWAS.best.txt | head -n 10

HG00096	HG00096	-0.000627027476
HG00097	HG00097	-0.00058856461
HG00099	HG00099	-0.000713382756
HG00100	HG00100	-0.000571751231
HG00101	HG00101	-0.000530233701
HG00102	HG00102	-0.000583765514
HG00103	HG00103	-0.000573780098
HG00104	HG00104	-0.000506927325
HG00106	HG00106	-0.000537212649
HG00108	HG00108	-0.000588234975


In [49]:
ls *

PRsice_Command_cygwin.txt  PRSice_linux    PRSice_mac.R
PRsice_Command_mac.txt     PRSice_linux.R  PRSice.R
PRsice_Command.txt         PRSice_mac      PRSice_win64.exe

Input:
1KG_EUR_QC.bed  1KG_EUR_QC.fam    RA_GWASmeta_European_PRSise.txt.gz
1KG_EUR_QC.bim  phenotype_RA.txt

lib:
data.table  getopt  optparse

Output:
1KG_EUR-RAGWAS_BARPLOT_2019-08-25.png        1KG_EUR-RAGWAS.log
1KG_EUR-RAGWAS_BARPLOT_2021-07-14.png        1KG_EUR-RAGWAS-PRSboxplot.png
1KG_EUR-RAGWAS.best                          1KG_EUR-RAGWAS.prsice
1KG_EUR-RAGWAS.best.txt                      1KG_EUR-RAGWAS.summary
1KG_EUR-RAGWAS_HIGH-RES_PLOT_2019-08-25.png  1KG_EUR_Sample.txt
1KG_EUR-RAGWAS_HIGH-RES_PLOT_2021-07-14.png  PlotPRSice.R

Output_2:
1KG_EUR-RAGWAS_BARPLOT_2023-12-11.png        1KG_EUR-RAGWAS.log
1KG_EUR-RAGWAS.best                          1KG_EUR-RAGWAS.prsice
1KG_EUR-RAGWAS.best.txt                      1KG_EUR-RAGWAS.summary
1KG_EUR-RAGWAS_HIGH-RES_PLOT_2023-12-11.png  PlotPRSice.R


In [57]:
#Rスクリプトの調整
cat Output/PlotPRSice.R -n

     1	#ディレクトリの移動
     2		rm(list = ls(all = TRUE))
     3		setwd("C:/SummerSchool/GenomeDataAnalysis5/PRSice/Output");
     4	
     5	
     6		PRS <- read.table("1KG_EUR-RAGWAS.best.txt", header=FALSE)[,3];
     7		Pop <- read.table("1KG_EUR_Sample.txt", header=FALSE)[,2];
     8	
     9		png(file="1KG_EUR-RAGWAS-PRSboxplot.png", width = 600, height = 600);
    10		par(mar = c(5, 5, 0.5, 0.5));
    11		boxplot(PRS~Pop, col=c(2,3,4,5,6), pch=16, xlab="1KG EUR sub-populations", ylab="Polygenic risk score for rheumatoid arthritis", cex.lab=2.0, cex.axis=2.0, cex.main=2.0);
    12		dev.off();
    13		
    14		anova(aov(PRS~Pop)) # P-value for Pop < 2.2x10^-16
    15		


In [51]:
cat Output/PlotPRSice.R | sed 1,5d

	PRS <- read.table("1KG_EUR-RAGWAS.best.txt", header=FALSE)[,3];
	Pop <- read.table("1KG_EUR_Sample.txt", header=FALSE)[,2];

	png(file="1KG_EUR-RAGWAS-PRSboxplot.png", width = 600, height = 600);
	par(mar = c(5, 5, 0.5, 0.5));
	boxplot(PRS~Pop, col=c(2,3,4,5,6), pch=16, xlab="1KG EUR sub-populations", ylab="Polygenic risk score for rheumatoid arthritis", cex.lab=2.0, cex.axis=2.0, cex.main=2.0);
	dev.off();
	
	anova(aov(PRS~Pop)) # P-value for Pop < 2.2x10^-16
	


In [52]:
echo 'setwd("Output_2")' > Output_2/PlotPRSice.R
cat Output/PlotPRSice.R | sed 1,5d >> Output_2/PlotPRSice.R
ls Output_2

1KG_EUR-RAGWAS_BARPLOT_2023-12-11.png        1KG_EUR-RAGWAS.log
1KG_EUR-RAGWAS.best                          1KG_EUR-RAGWAS.prsice
1KG_EUR-RAGWAS.best.txt                      1KG_EUR-RAGWAS.summary
1KG_EUR-RAGWAS_HIGH-RES_PLOT_2023-12-11.png  PlotPRSice.R


In [53]:
cat Output_2/PlotPRSice.R

setwd("Output_2")
	PRS <- read.table("1KG_EUR-RAGWAS.best.txt", header=FALSE)[,3];
	Pop <- read.table("1KG_EUR_Sample.txt", header=FALSE)[,2];

	png(file="1KG_EUR-RAGWAS-PRSboxplot.png", width = 600, height = 600);
	par(mar = c(5, 5, 0.5, 0.5));
	boxplot(PRS~Pop, col=c(2,3,4,5,6), pch=16, xlab="1KG EUR sub-populations", ylab="Polygenic risk score for rheumatoid arthritis", cex.lab=2.0, cex.axis=2.0, cex.main=2.0);
	dev.off();
	
	anova(aov(PRS~Pop)) # P-value for Pop < 2.2x10^-16
	


In [59]:
#必要ファイルの準備
cp Output/1KG_EUR_Sample.txt Output_2/

In [60]:
#Rスクリプトの実行
Rscript Output_2/PlotPRSice.R

null device 
          1 
Analysis of Variance Table

Response: PRS
           Df     Sum Sq    Mean Sq F value    Pr(>F)    
Pop         4 6.4407e-07 1.6102e-07  33.237 < 2.2e-16 ***
Residuals 376 1.8215e-06 4.8440e-09                      
---
Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1


In [62]:
ls Output_2

1KG_EUR-RAGWAS_BARPLOT_2023-12-11.png        1KG_EUR-RAGWAS-PRSboxplot.png
1KG_EUR-RAGWAS.best                          1KG_EUR-RAGWAS.prsice
1KG_EUR-RAGWAS.best.txt                      1KG_EUR-RAGWAS.summary
1KG_EUR-RAGWAS_HIGH-RES_PLOT_2023-12-11.png  1KG_EUR_Sample.txt
1KG_EUR-RAGWAS.log                           PlotPRSice.R


In [63]:
#画像表示の準備
cp Output_2/1KG_EUR-RAGWAS-PRSboxplot.png ~/hmiwa/hmiwa-lab/d_20231023/05_231211_RePRSTrials/pic

<img src="pic/1KG_EUR-RAGWAS-PRSboxplot.png" width="49%">

### 東アジア集団GWAS（BaseData）のQC

まずややこしいのでコード格納ディレクトリとストレージディレクトリをそれぞれ変数にあてておく

In [2]:
gitdir=~/hmiwa/hmiwa-lab/d_20231023/05_231211_RePRSTrials
ls $gitdir

strdir=~/hmiwa1/storage/hmiwa-lab/05_231211_RePRSTrials
ls $strdir

05_231211_RePRSTrials.ipynb  Documents  jupyter_venv5  pic  test.ipynb
Basedata
StatGenSummerSchool2023_GenomeDataAnalysis5
StatGenSummerSchool2023_GenomeDataAnalysis5.zip


In [78]:
cd $strdir
mkdir -p Basedata
ls

Basedata
StatGenSummerSchool2023_GenomeDataAnalysis5
StatGenSummerSchool2023_GenomeDataAnalysis5.zip


#### データのダウンロード

論文はこれを使う。

https://www-nature-com.utokyo.idm.oclc.org/articles/s41588-017-0011-x

In [79]:
cd $strdir/Basedata
wget https://portals.broadinstitute.org/collaboration/giant/images/7/78/BMI_Eastern_Asian.fmt.gzip

--2023-12-11 19:14:33--  https://portals.broadinstitute.org/collaboration/giant/images/7/78/BMI_Eastern_Asian.fmt.gzip
Resolving portals.broadinstitute.org (portals.broadinstitute.org)... 69.173.68.142
Connecting to portals.broadinstitute.org (portals.broadinstitute.org)|69.173.68.142|:443... connected.
HTTP request sent, awaiting response... 200 OK
Length: 12268615 (12M)
Saving to: ‘BMI_Eastern_Asian.fmt.gzip’

BMI_Eastern_Asian.f 100%[===================>]  11.70M  5.61MB/s    in 2.1s    

2023-12-11 19:14:36 (5.61 MB/s) - ‘BMI_Eastern_Asian.fmt.gzip’ saved [12268615/12268615]



In [80]:
ls $strdir/Basedata

BMI_Eastern_Asian.fmt.gzip


In [81]:
head BMI_Eastern_Asian.fmt.gzip

CHR	POS	REF	ALT	SNPNAME	EAS_MAF	ExAC_EAS_MAF	beta	se	Pvalue
10	100003875	G	A	rs148923672	-	A:0	NA	Inf	NA
10	100008694	C	G	rs200213822	-	G:0	1.6	1	0.1
10	100010909	C	T	rs141069497	T:0.0139	T:0.006614	-0.022	0.1	0.83
10	100011387	C	T	rs199733612	-	T:0,G:0	1.1	1	0.27
10	100011442	C	T	rs74153530	T:0	T:0,G:0	0.66	0.59	0.26
10	100013325	C	T	rs145270262	-	T:0	NA	Inf	NA
10	100013367	C	T	rs199557224	-	T:0	NA	Inf	NA
10	100013382	A	G	rs141995277	G:0	G:0	NA	Inf	NA
10	100013418	T	G	rs148260647	-	G:0	0.41	1	0.68


#### QCのチェックリスト
* Heritability Check
* Effect Allele
* File Transfer
* Genome Build
* Standard GWAS QC
* Mismatching SNPs
* Duplicate SNPs
* Ambiguous SNPs
* Sex Chromosomes
* Sample overlap
* Relatedness

##### Heritability Check

今回はスキップさせてくれ…　たぶん大丈夫だとは思うのだが。

（この前提が崩れるとそもそもこれが使えない。そういうものは公開されてないと思う。）

##### Effect Allele

MAFでフィルタリングするときに同時にチェックします。

##### File Transfer

ファイルの損傷チェック。とりあえずスキップ。

##### Genome Build：hg19/hg38が合っているかのチェック

In [60]:
head -n 2 BMI_Eastern_Asian.fmt.gzip

CHR	POS	REF	ALT	SNPNAME	EAS_MAF	ExAC_EAS_MAF	beta	se	Pvalue
10	100003875	G	A	rs148923672	-	A:0	NA	Inf	NA


https://www.ncbi.nlm.nih.gov/snp/?term=rs148923672

上記を確認すると、

    10:98244118 (GRCh38)
    10:100003875 (GRCh37)
    
とあるので、このデータはGRCh37で間違いなさそう。

##### Standard GWAS QC：MAFやINFOが低いSNPは除去した方が良い！

In [61]:
cd $strdir/Basedata

In [62]:
head BMI_Eastern_Asian.fmt.gzip | \
sed -e s/A://g -e s/T://g -e s/G://g -e s/C://g | \
grep -v ,

CHR	POS	REF	ALT	SNPNAME	EAS_MAF	ExAC_EAS_MAF	beta	se	Pvalue
10	100003875	G	A	rs148923672	-	0	NA	Inf	NA
10	100008694	C	G	rs200213822	-	0	1.6	1	0.1
10	100010909	C	T	rs141069497	0.0139	0.006614	-0.022	0.1	0.83
10	100013325	C	T	rs145270262	-	0	NA	Inf	NA
10	100013367	C	T	rs199557224	-	0	NA	Inf	NA
10	100013382	A	G	rs141995277	0	0	NA	Inf	NA
10	100013418	T	G	rs148260647	-	0	0.41	1	0.68


In [63]:
head BMI_Eastern_Asian.fmt.gzip | \
sed -e s/A://g -e s/T://g -e s/G://g -e s/C://g | \
grep -v , | \
awk 'NR==1 || ($6 > 0.01) {print}'

CHR	POS	REF	ALT	SNPNAME	EAS_MAF	ExAC_EAS_MAF	beta	se	Pvalue
10	100010909	C	T	rs141069497	0.0139	0.006614	-0.022	0.1	0.83


In [64]:
head BMI_Eastern_Asian.fmt.gzip | \
grep -v , 

CHR	POS	REF	ALT	SNPNAME	EAS_MAF	ExAC_EAS_MAF	beta	se	Pvalue
10	100003875	G	A	rs148923672	-	A:0	NA	Inf	NA
10	100008694	C	G	rs200213822	-	G:0	1.6	1	0.1
10	100010909	C	T	rs141069497	T:0.0139	T:0.006614	-0.022	0.1	0.83
10	100013325	C	T	rs145270262	-	T:0	NA	Inf	NA
10	100013367	C	T	rs199557224	-	T:0	NA	Inf	NA
10	100013382	A	G	rs141995277	G:0	G:0	NA	Inf	NA
10	100013418	T	G	rs148260647	-	G:0	0.41	1	0.68


In [65]:
head BMI_Eastern_Asian.fmt.gzip | \
grep -v , | \
sed -e s/:/"\t"/g -e s/-/-"\t"-/g

CHR	POS	REF	ALT	SNPNAME	EAS_MAF	ExAC_EAS_MAF	beta	se	Pvalue
10	100003875	G	A	rs148923672	-	-	A	0	NA	Inf	NA
10	100008694	C	G	rs200213822	-	-	G	0	1.6	1	0.1
10	100010909	C	T	rs141069497	T	0.0139	T	0.006614	-	-0.022	0.1	0.83
10	100013325	C	T	rs145270262	-	-	T	0	NA	Inf	NA
10	100013367	C	T	rs199557224	-	-	T	0	NA	Inf	NA
10	100013382	A	G	rs141995277	G	0	G	0	NA	Inf	NA
10	100013418	T	G	rs148260647	-	-	G	0	0.41	1	0.68


In [66]:
echo -e 'CHR\tPOS\tREF\tALT\tSNPNAME\tEAS\tEAS_MAF\tExAC_EAS\tExAC_EAS_MAF\tbeta\tse\tPvalue'
head BMI_Eastern_Asian.fmt.gzip | \
grep -v , | \
sed -e s/:/"\t"/g -e s/-/-"\t"-/g

CHR	POS	REF	ALT	SNPNAME	EAS	EAS_MAF	ExAC_EAS	ExAC_EAS_MAF	beta	se	Pvalue


CHR	POS	REF	ALT	SNPNAME	EAS_MAF	ExAC_EAS_MAF	beta	se	Pvalue
10	100003875	G	A	rs148923672	-	-	A	0	NA	Inf	NA
10	100008694	C	G	rs200213822	-	-	G	0	1.6	1	0.1
10	100010909	C	T	rs141069497	T	0.0139	T	0.006614	-	-0.022	0.1	0.83
10	100013325	C	T	rs145270262	-	-	T	0	NA	Inf	NA
10	100013367	C	T	rs199557224	-	-	T	0	NA	Inf	NA
10	100013382	A	G	rs141995277	G	0	G	0	NA	Inf	NA
10	100013418	T	G	rs148260647	-	-	G	0	0.41	1	0.68


In [67]:
echo -e 'CHR\tPOS\tREF\tALT\tSNPNAME\tEAS\tEAS_MAF\tExAC_EAS\tExAC_EAS_MAF\tbeta\tse\tPvalue'
head BMI_Eastern_Asian.fmt.gzip | \
grep -v , | \
sed -e s/:/"\t"/g -e s/-/-"\t"-/g | \
sed 1d

CHR	POS	REF	ALT	SNPNAME	EAS	EAS_MAF	ExAC_EAS	ExAC_EAS_MAF	beta	se	Pvalue


10	100003875	G	A	rs148923672	-	-	A	0	NA	Inf	NA
10	100008694	C	G	rs200213822	-	-	G	0	1.6	1	0.1
10	100010909	C	T	rs141069497	T	0.0139	T	0.006614	-	-0.022	0.1	0.83
10	100013325	C	T	rs145270262	-	-	T	0	NA	Inf	NA
10	100013367	C	T	rs199557224	-	-	T	0	NA	Inf	NA
10	100013382	A	G	rs141995277	G	0	G	0	NA	Inf	NA
10	100013418	T	G	rs148260647	-	-	G	0	0.41	1	0.68


In [68]:
echo -e 'CHR\tPOS\tREF\tALT\tSNPNAME\tEAS\tEAS_MAF\tExAC_EAS\tExAC_EAS_MAF\tbeta\tse\tPvalue' > BMI_Eastern_Asian.newcol.txt
cat BMI_Eastern_Asian.fmt.gzip | \
grep -v , | \
sed -e s/:/"\t"/g -e s/-/-"\t"-/g | \
sed 1d >> BMI_Eastern_Asian.newcol.txt 

In [69]:
cat BMI_Eastern_Asian.newcol.txt | \
awk 'NR==1 || ($7 > 0.01) {print}' | wc

  31194  389695 1870772


In [70]:
cat BMI_Eastern_Asian.newcol.txt | \
awk 'NR==1 || ($7 > 0.01) && ($4 == $6) {print}' | wc

  31170  389376 1869585


In [71]:
cat BMI_Eastern_Asian.newcol.txt | \
awk 'NR==1 || ($7 > 0.01) && ($4 == $6) {print}' > BMI_Eastern_Asian.txt

In [72]:
head BMI_Eastern_Asian.txt

CHR	POS	REF	ALT	SNPNAME	EAS	EAS_MAF	ExAC_EAS	ExAC_EAS_MAF	beta	se	Pvalue
10	100010909	C	T	rs141069497	T	0.0139	T	0.006614	-	-0.022	0.1	0.83
10	100017453	T	G	rs1983864	G	0.6081	G	0.5682	-	-0.031	0.015	0.044
10	100017532	G	A	rs77858766	A	0.0169	A	0.01781	-	-0.016	0.058	0.79
10	100017553	G	T	rs11189525	T	0.0129	T	0.007404	-	-0.014	0.13	0.91
10	100020880	C	T	rs33995374	T	0.1062	T	0.1163	0.077	0.061	0.21
10	100044136	G	A	rs4919199	A	0.9544	-	-	0.0076	0.036	0.83
10	100144782	C	T	rs2296441	T	0.4425	T	0.42	0.012	0.015	0.42
10	100148004	T	C	rs2182169	C	0.5615	-	-	0.014	0.015	0.37
10	100148176	A	G	rs2147896	G	0.5833	G	0.5596	0.0052	0.015	0.73


In [73]:
ls -l BMI_Eastern_Asian.txt
wc BMI_Eastern_Asian.txt

-rw-rw-r-- 1 hmiwa hmiwa 1869585 12月 11 21:03 BMI_Eastern_Asian.txt


  31170  389376 1869585 BMI_Eastern_Asian.txt


##### Mismatching SNPs：ターゲットデータで対立遺伝子が一致しているかどうかのチェック。

ターゲットデータがないとチェックできないのでここはスキップ。

##### Duplicate SNPs

In [74]:
cat BMI_Eastern_Asian.txt |\
awk '{seen[$5]++; if(seen[$5]==1){print}}' > BMI_Eastern_Asian.nodup.txt

In [75]:
ls -l BMI_Eastern_Asian.nodup.txt
wc BMI_Eastern_Asian.nodup.txt

-rw-rw-r-- 1 hmiwa hmiwa 1869585 12月 11 21:03 BMI_Eastern_Asian.nodup.txt


  31170  389376 1869585 BMI_Eastern_Asian.nodup.txt


##### Ambiguous SNPs

In [76]:
head BMI_Eastern_Asian.nodup.txt

CHR	POS	REF	ALT	SNPNAME	EAS	EAS_MAF	ExAC_EAS	ExAC_EAS_MAF	beta	se	Pvalue
10	100010909	C	T	rs141069497	T	0.0139	T	0.006614	-	-0.022	0.1	0.83
10	100017453	T	G	rs1983864	G	0.6081	G	0.5682	-	-0.031	0.015	0.044
10	100017532	G	A	rs77858766	A	0.0169	A	0.01781	-	-0.016	0.058	0.79
10	100017553	G	T	rs11189525	T	0.0129	T	0.007404	-	-0.014	0.13	0.91
10	100020880	C	T	rs33995374	T	0.1062	T	0.1163	0.077	0.061	0.21
10	100044136	G	A	rs4919199	A	0.9544	-	-	0.0076	0.036	0.83
10	100144782	C	T	rs2296441	T	0.4425	T	0.42	0.012	0.015	0.42
10	100148004	T	C	rs2182169	C	0.5615	-	-	0.014	0.015	0.37
10	100148176	A	G	rs2147896	G	0.5833	G	0.5596	0.0052	0.015	0.73


In [77]:
cat BMI_Eastern_Asian.nodup.txt |\
awk '!( ($3=="A" && $4=="T") || \
        ($3=="T" && $4=="A") || \
        ($3=="G" && $4=="C") || \
        ($3=="C" && $4=="G")) {print}' > BMI_Eastern_Asian.QC.txt

In [78]:
ls -l BMI_Eastern_Asian.QC.txt
wc BMI_Eastern_Asian.QC.txt

-rw-rw-r-- 1 hmiwa hmiwa 1707833 12月 11 21:04 BMI_Eastern_Asian.QC.txt


  28519  356256 1707833 BMI_Eastern_Asian.QC.txt


##### Sex Chromosomes

やりようがないのでスルー

##### Sample overlap
オセアニア集団とは絶対重ならないのでOK

##### Relatedness

本来はやるべきだが今回は一旦スキップ

### Tonga集団サンプル（TargetData）のQC